src/component-library/Stack/Stack.tsx

```diff
@@ -1,18 +1,21 @@
+import { HTMLAttributes } from 'react';
+
 import { StackContainer } from './Stack.style';
 
-interface StackProps {
-  children: React.ReactNode;
-  className?: string;
+type Props = {
+  as?: keyof JSX.IntrinsicElements;
```

Comment:
@danielsimao what's the use case for this? I like the idea of the stack being a very simple component so that we don't end up with something like `<Stack as="form" onSubmit={handleSubmit}>`. That might mean we end up with an additional nested element, but I prefer that as it gives us a slightly more obvious separation of concerns.

Reply:
My idea was in case you want to make the stack a `section`. This is a component used for grouping a set of components and a `section` is used to group a set of `articles` for example.

But now that I put some thought into it, you can do that by using `role`. Thank you for the review. I will update it.

Reply:
Also, we should consider adding `direction` prop in the future, in case we want a row stack.

Reply:
Yes - if this comes up we can add it in. I've never found a row stack as useful as a column stack, but every project's different :)